src/pages/iv-materials-tfr/index.js

```diff
@@ -0,0 +1,265 @@
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50, params } = options
+
+    const response = await getRequest({
+      extension: InventoryRepository.MaterialsTransfer.qry,
+      parameters: `_filter=&_size=30&_startAt=${_startAt}&_sortBy=recordId desc&_pageSize=${_pageSize}&_params=${params || ''}`
+    })
+
+    if (response && response?.list) {
+      response.list = response?.list?.map(item => ({
+        ...item,
+        isVerified: item?.isVerified === null ? false : item?.isVerified
+      }))
+    }
+
+    return { ...response, _startAt: _startAt }
+  }
+
+  async function fetchWithFilter({ filters, pagination }) {
+    if (filters?.qry) {
+      return await getRequest({
+        extension: InventoryRepository.MaterialsTransfer.snapshot,
+        parameters: `_filter=${filters.qry}`
+      })
+    } else {
+      return fetchGridData({ _startAt: pagination._startAt || 0, params: filters?.params })
+    }
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    filterBy,
+    clearFilter,
+    paginationParameters,
+    refetch,
+    access,
+    invalidate
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: InventoryRepository.MaterialsTransfer.qry,
+    datasetId: ResourceIds.MaterialsTransfer,
+    filter: {
+      filterFn: fetchWithFilter
+    }
+  })
+
+  const columns = [
+    {
+      field: 'reference',
+      headerName: _labels.reference,
+      flex: 1
+    },
+    {
+      field: 'date',
+      headerName: _labels.date,
+      flex: 1,
+      type: 'date'
+    },
+
+    {
+      field: 'fromSiteRef',
+      headerName: _labels.fromSiteRef,
+      flex: 1
+    },
+    {
+      field: 'fromSiteName',
+      headerName: _labels.fromSite,
+      flex: 1
+    },
+    {
+      field: 'toSiteRef',
+      headerName: _labels.toSiteRef,
+      flex: 1
+    },
+    {
+      field: 'toSiteName',
+      headerName: _labels.toSite,
+      flex: 1
+    },
+    {
+      field: 'closedDate',
+      headerName: _labels.shippingDate,
+      flex: 1,
+      type: 'date'
+    },
+    {
+      field: 'receivedDate',
+      headerName: _labels.receivedDate,
+      flex: 1,
+      type: 'date'
+    },
+    {
+      field: 'pcs',
+      headerName: _labels.pcs,
+      flex: 1,
+      type: 'number'
+    },
+    {
+      field: 'qty',
+      headerName: _labels.qty,
+      flex: 1,
+      type: 'number'
+    },
+    {
+      field: 'statusName',
+      headerName: _labels.status,
+      flex: 1
+    },
+    {
+      field: 'wipName',
+      headerName: _labels.wip,
+      flex: 1
+    },
+    {
+      field: 'printStatusName',
+      headerName: _labels.print,
+      flex: 1
+    },
+    {
+      field: 'notes',
+      headerName: _labels.notes,
+      flex: 1
+    },
+
+    {
+      field: 'isVerified',
+      headerName: _labels.isVerified,
+      flex: 1
+    }
+  ]
+
+  const add = () => {
+    openForm()
```

Comment:
why u are not using  proxyAction() on add? to check if it's bind to system function or not